### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by nedzadarek** [HackerPoet/NonEuclidean#11]
I am using pre-built exe. 
The more I use it the more it takes ram. The ram usage is increasing rapidly. First level it's just takes ~50 mb per thing, The next levels starts to take hundreds (100+) ram. After few changes I got ~1GB. Last ram usage was ~1.8 GB and my laptop was frozen. I had to "hard shutdown" it (hold power button for some time).
My laptop:
64x bit Win 8.1 4gb ram.

**Comment by Chaosminecraft** [comment]
How can I run this program?

**Comment by Fran314** [comment]
I've also found some pretty serious memory leak issues. I'm using the VisualStudio version of the software and I'm building it myself, I just added the glew-2.1.0. I've found that each time a level is loaded, it increases the amount of RAM used, so I think the issue might be somewhere around the loading level system.

Here's the RAM usage as shown from the VisualStudio Diagnostic Tool
First level: 388 MB
2nd: 565 MB
3rd: 794 MB
4th: 1 GB
5th: 1,2 GB
6th: 1,5 GB
7th: 2 GB

This is what happens if you load the levels one after the other. During the level the RAM usage stays perfectly constant, and only spikes on the new level. If you load a level once again after having already played it, it still increases the RAM usage

**Comment by ElijahZAwesome** [comment]
Yeah, I noticed that too. He (or someone) needs to make the game unload the previous level when loading a new one.

**Comment by ehx-v1** [comment]
> 
> 
>  If you load a level once again after having already played it, it still increases the RAM usage

That sounds like it's a straight up case of not freeing resources that should be freed. Maybe take a look with Valgrind?

**Comment by Chaosminecraft** [comment]
how can i run this program?

**Comment by Chaosminecraft** [comment]
Hi.

**Comment by angebrajuka** [comment]
Hey, I think I narrowed down the memory leak to the mesh class, don't know if this helps anyone or if anyone else is trying to fix this

**Comment by ehx-v1** [comment]
@Chaosminecraft When you download the repository, there's some files directly in the ZIP, and most of the stuff is in the folder `NonEuclidean`. Among that bunch, there's also a `NonEuclidean.exe`, simply launch that.
Assuming you're on Windows. On Linux you'll have to launch it with Wine, and I think there was something similar to Wine for MacOS as well.

**Comment by Chaosminecraft** [comment]
@ehx-v1 thanks, I did not explore the folders yet, I'll look if I still have that or I download it again.
Cheers 
Chaosminecraft
